src/hotspot/share/opto/compile.hpp

```diff
@@ -521,6 +521,7 @@ class Compile : public Phase {
   Matcher*              _matcher;               // Engine to map ideal to machine instructions
   PhaseRegAlloc*        _regalloc;              // Results of register allocation.
   RegMask               _FIRST_STACK_mask;      // All stack slots usable for spills (depends on frame layout)
+  ResourceArea          _regmask_arena;         // Holds dynamically allocated extensions of short-lived register masks
```

Comment:
If they are short-lived ... then why not just resource allocate them? Is there a conflict? Would it be good to describe that somewhere?

Reply:
And these conflicts cannot be resolved? Can you bring an example that is too much effort?

Reply:
One example is the `ResourceMark` at the top of `PhaseCFG::sched_call` (added in this changeset) which conflicts with the `ResourceMark` at the top of `PhaseCFG::global_code_motion`. Specifically, the conflict is at the `VectorSet` named `visited` in `PhaseCFG::global_code_motion` and named `next_call` in `PhaseCFG::sched_call`. I don't see a trivial resolution, but I'm open to suggestions. There are loops in between the resource marks, so memory consumption will potentially increase significantly if we simply remove the inner resource mark.